### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by KhoonSun47**
Update Documentation:

Changes made:
- DeveloperGuide.md: Update manual instructions for adding seller,
loading data, saving data

### CODE REVIEW

**Review by redcolorbicycle — APPROVED** [review]

**Review by zengzihui — APPROVED** [review]
lgtm

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -909,51 +909,66 @@ testers are expected to do more *exploratory* testing.
 
 [//]: # (@@author KhoonSun47)
 ### Loading Data
+1. **Dealing with Missing Data Folder**
 
-1. **Dealing with Missing Data Folder/ Missing Data File (with Data Folder)**
+    1. **Test case:** Manually delete the `data` folder.
```
Comment (felixchanyy) [inline docs/DeveloperGuide.md:914]: Missing a <br>

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -909,51 +909,66 @@ testers are expected to do more *exploratory* testing.
...
+       **Expected:** The application automatically populates EstateEase with sample data, displaying buyers and sellers where sellers are associated with houses. <br><br>
 
-    1. **Loading Data:**
+3. **Dealing with Corrupted Data Files**
 
-        - To test the application's response to a missing data folder or `addressbook.json` file, manually delete the `data` folder or the `addressbook.json` file from it.
-        - The application should automatically populate EstateEase with sample data, displaying buyers and sellers, where sellers are associated with houses. <br>
-        <br>
-1. **Dealing with Corrupted Data Files**
+    1. **Test case:** Duplicate a buyer or seller's details (name, phone, and email) and use them for the opposite role (e.g., use a buyer's details for a seller or vice versa).
```
Comment (felixchanyy) [inline docs/DeveloperGuide.md:924]: Same as above

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1162,21 +1177,24 @@ similar to the test cases found in `Editing Seller Details`.
 
 [//]: # (@@author KhoonSun47)
 ### Saving Data
+**Prerequisites:**
+- Execute a valid command.
+
+1. **Dealing with Missing Data Folder**
+
+    1. **Test case** (Missing data folder): Manually delete the `data` folder.
```
Comment (felixchanyy) [inline docs/DeveloperGuide.md:1185]: same as above

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -909,51 +909,66 @@ testers are expected to do more *exploratory* testing.
...
+   4. **Test case:** `addSeller n/Carl Lim Jovi Rato p/9876-9999 e/carllimjovirato@example.com type/Hdb street/Toa Payoh Ave 10 blk/312 level/22 unitNo/39 postal/528978 price/9999999` <br>
+      **Expected:** No seller is added. Error indicating phone numbers should only contain numbers. <br><br>
+
+   5. **Test case:** `addSeller n/Carl Lim Jovi Rato p/98769999 e/carllimjovirato-example.com type/Hdb street/Toa Payoh Ave 10 blk/312 level/22 unitNo/39 postal/528978 price/9999999` <br>
+      **Expected:** No seller is added. Error indicating that email should be in the format of should be of the format local-part@domain. <br><br>
 
 3. **Invalid (Duplicate)**
-   1. **Test case** (Duplicate seller -> seller and buyer cannot be same name): `addSeller n/John Carl 1 p/98765432 e/johncarl1@example.com type/Landed street/Clementi Ave 2 unitNo/25 postal/578578 price/10000`
-      **Expected**: No seller is added. Error indicating that the person already existed in the data. <br><br>
+   1. **Test case:** `addSeller n/John Carl 1 p/98765432 e/johncarl1@example.com type/Landed street/Clementi Ave 2 unitNo/25 postal/578578 price/10000`
```
Comment (felixchanyy) [inline docs/DeveloperGuide.md:967]: Missing a `<br>`

**Review by felixchanyy — APPROVED** [review]
LGTM

**Review by lokidoki102 — APPROVED** [review]

**Pull request merged**